ESOF-docs/report_1.md

```diff
@@ -0,0 +1,61 @@
+# 1st Report - ESOF
+
+We are 4 students of the Engineering Faculty of the University of Oporto, and we're studying this project for the course of Software Engineering.
+
+## Index
+
+1. [RUST - Programming Language](#rust---programming-language)
+2. [Project's Activity](#projects-activity)
+3. [Developing Paradigm](#developing-paradigm)
+    1. [Paradigm problems](#paradigm-problems)
+    2. TAD (Test after develpment) vs TDD (Test driven development)
+
+## RUST - Programming Language
+
+Rust is a fast systems programming language that guarantees memory safety and offers painless concurrency (no data races). It does not employ a garbage collector and has minimal runtime overhead.
+
+***
+
+## Project's Activity
+##### _Since 13/06/2010 Until 28/09/2015_
+
+This project rellies on 1.164 contributors with a total of 46.757 commits.
+In the last month it had 358 active pull requests. 322 of those are Merged pull request, and the other 36 are Proposed.
+Regarding Issues there were 162 new ones and 259 closed.
+
+***
+
+## Developing Paradigm
+
+Rust programming language uses **Test Driven Development - TDD** as its developing paradigm.
+
+    The rhythm of Test-Driven Development (TDD) can be summed up as follows:
+    1. Quickly add a test.
+    2. Run all tests and see the new one fail.
+    3. Make a little change.
+    4. Run all tests and see them all succeed.
+    5. Refactor to remove duplication.
```

Comment:
FWIW we don't particularly follow TDD. I don't think that's actually possible to even enforce at a project or language level; it's an individual development paradigm, and not something that can really be enforced in a distributed manner. In fact, the TDD I know would be quite brutal to apply to Rust because of it's massive build cycles. Regardless, all we get are patches, and we don't really care *how* they were developed. We do however strongly favour changes that come with associated tests, because tests are great!